### PR TITLE
Ec 418 Core - create metric for timing per sync or for timing per block while syncing

### PR DESCRIPTION
**Pull request by GusevTimofey**

### CODE REVIEW

File: src/main/scala/encry/network/EncryNodeViewSynchronizer.scala [hunk 1]
```diff
@@ -41,6 +42,9 @@ class EncryNodeViewSynchronizer extends Actor with Logging {
     context.system.eventStream.subscribe(self, classOf[NodeViewChange])
     context.system.eventStream.subscribe(self, classOf[ModificationOutcome])
     nodeViewHolder ! GetNodeViewChanges(history = true, state = false, vault = false, mempool = true)
+    if (settings.network.knownPeers.nonEmpty && !settings.node.offlineGeneration && settings.influxDB.isDefined &&
```
Comment (GRYE) [inline src/main/scala/encry/network/EncryNodeViewSynchronizer.scala]: Why knownPeers should be non-empty?

File: src/main/scala/encry/network/EncryNodeViewSynchronizer.scala [hunk 2]
```diff
@@ -41,6 +42,9 @@ class EncryNodeViewSynchronizer extends Actor with Logging {
...
+      settings.levelDb.exists(x => !x.enableRestore) && settings.postgres.exists(x => !x.enableRestore))
```
Comment (GRYE) [inline src/main/scala/encry/network/EncryNodeViewSynchronizer.scala]: I think you wanted to use `forall` here

File: src/main/scala/encry/network/EncryNodeViewSynchronizer.scala [hunk 3]
```diff
@@ -41,6 +42,9 @@ class EncryNodeViewSynchronizer extends Actor with Logging {
...
+      context.actorSelection("/user/statsSender") ! StartRecoveryFromNetwork(System.currentTimeMillis())
```
Comment (GRYE) [inline src/main/scala/encry/network/EncryNodeViewSynchronizer.scala]: Use `System.nanoTime`

File: src/main/scala/encry/network/EncryNodeViewSynchronizer.scala [hunk 4]
```diff
@@ -41,6 +42,9 @@ class EncryNodeViewSynchronizer extends Actor with Logging {
     context.system.eventStream.subscribe(self, classOf[NodeViewChange])
     context.system.eventStream.subscribe(self, classOf[ModificationOutcome])
     nodeViewHolder ! GetNodeViewChanges(history = true, state = false, vault = false, mempool = true)
+    if (!settings.node.offlineGeneration && settings.influxDB.isDefined &&
+      settings.levelDb.forall(x => !x.enableRestore) && settings.postgres.exists(x => !x.enableRestore))
```
Comment (GRYE) [inline src/main/scala/encry/network/EncryNodeViewSynchronizer.scala]: formal for postgres settings

File: src/main/scala/encry/network/ModifiersHolder.scala [hunk 5]
```diff
@@ -69,11 +75,17 @@ class ModifiersHolder extends PersistentActor with Logging {
       val blocksToSend: Seq[Block] = completedBlocks.take(settings.levelDb
         .map(_.batchSize)
         .getOrElse(throw new RuntimeException("batchsize not specified"))).values.toSeq
-
+      if (completedBlocks.values.size < settings.levelDb.map(_.batchSize).getOrElse(1000)
```
Comment (Bromel777) [inline src/main/scala/encry/network/ModifiersHolder.scala]: magic number

File: src/main/scala/encry/network/ModifiersHolder.scala [hunk 6]
```diff
@@ -69,11 +75,17 @@ class ModifiersHolder extends PersistentActor with Logging {
...
+        && settings.influxDB.isDefined && !isSentFinishRecovery) {
+        isSentFinishRecovery = true
+        context.actorSelection("/user/statsSender") ! FinishRecoveryFromLevelDb(System.currentTimeMillis())
+        println("done")
```
Comment (Bromel777) [inline src/main/scala/encry/network/ModifiersHolder.scala]: remove println

File: src/main/scala/encry/network/ModifiersHolder.scala [hunk 7]
```diff
@@ -20,6 +21,7 @@ class ModifiersHolder extends PersistentActor with Logging {
   var payloads: Map[String, (Payload, Int)] = Map.empty
   var nonCompletedBlocks: Map[String, String] = Map.empty
   var completedBlocks: SortedMap[Int, Block] = SortedMap.empty
+  var isSentFinishRecovery = false
```
Comment (Bromel777) [inline src/main/scala/encry/network/ModifiersHolder.scala]: you can use context.become

**Pull request closed without merging**